dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart

```diff
@@ -50,7 +55,36 @@ final Map<String, RecorderFactory> benchmarks = <String, RecorderFactory>{
     BenchTextCachedLayout.canvasBenchmarkName: () => BenchTextCachedLayout(useCanvas: true),
     BenchBuildColorsGrid.domBenchmarkName: () => BenchBuildColorsGrid.dom(),
     BenchBuildColorsGrid.canvasBenchmarkName: () => BenchBuildColorsGrid.canvas(),
-  }
+  },
+
+  // The following benchmarks are for the Flutter Gallery.
+  '${_galleryBenchmarkPrefix}_studies_perf': () => GalleryRecorder(
+    benchmarkName: '${_galleryBenchmarkPrefix}_studies_perf',
+    reporter: _client.reportDemo,
+    reportError: _client.reportError,
```

Comment:
We should make these available to all benchmarks, and in a way that each benchmark doesn't have to do this plumbing. For example, we could have top-level functions `printToConsole` and `printErrorToConsole` that any recorder can call. During the setup (first thing inside `_runBenchmark`) this file would redirect those function to `_client.reportDemo`, etc.

Reply:
## Print reports

**Is it good practice to use top-level functions that can be reset?**

I like this solution, and I can implement it if there's no restriction.

My idea is adding `printToConsole` to `recorder.dart` as a top-level function, and setting it to `_client.printToConsole` in the `main` function in `lib/web_benchmarks.dart`.

## Errors

I just realized that `web_benchmarks.dart` already handles errors:
https://github.com/flutter/flutter/blob/77ea2da7464a0c2801d9aad4214483b43b235f5b/dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart#L96

I would remove `reportError` for the `GalleryRecorder`s, and let `GalleryRecorder` raise errors directly.

Reply:
> Is it good practice to use top-level functions that can be reset?

It's not my favorite, but that's the pattern that Flutter uses quite extensively.

> I would remove reportError for the GalleryRecorders, and let GalleryRecorder raise errors directly.

SGTM. Do you need to pass it to `GalleryRecorder`?

Reply:
> SGTM. Do you need to pass it to `GalleryRecorder`?

I don't think we need to pass `_client.reportError` to `GalleryRecorder`. Errors are automatically caught by `web_benchmarks.dart`, so when I need to report an error, I'd simply raise it from the `GalleryRecorder`.

Reply:
@yjbanov 

I worked on this a bit more and realized:
- If any **async** exceptions were thrown inside any of these `Recorder`s, it wouldn't be caught by the `catch` statement in [lib/web_benchmarks.dart](https://github.com/flutter/flutter/blob/master/dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart#L96)

This happened with `GalleryRecorder` so I noticed it. It didn't catch the exception, so the program inside Chrome was paused, and the driver (tasks/web_benchmarks.dart) was always waiting for an answer.

A possible solution is wrapping the entire `try { ... }` block in a zone, so that **any** exception in this zone, async or not, are caught and reported.
(I think you mentioned zones earlier.)

If we use zones, it would make `printToConsole` unnecessary too, since we can also use a print handler on zones.

What do you think?

Reply:
I think it's a great idea! It will catch the cases when some asynchronous error affects what we render, which could lead to false benchmark numbers.

Reply:
Replaced `try … catch` with zones. I did an experiment, and it turns out zones can catch synchronous errors too.